### PR TITLE
Add postsubmit job updating knative prow deployment

### PR DESCRIPTION
**Pull request by chaodaiG**
/hold
Will need to give the service account permission to do so

/assign @cjwagner @fejta 
/cc @chizhg

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml [hunk 1]
```diff
@@ -73,6 +73,26 @@ postsubmits:
         - prow/oss
         - deploy
         - deploy-build
+  - name: post-oss-test-infra-deploy-knative-prow
+    cluster: test-infra-trusted
+    run_if_changed: '^prow/knaitve/cluster/|^prow/oss/serviceaccounts/'
```
Comment (chizhg) [inline prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml:78]: Typo, `knaitve` => `knative`

Comment (chaodaiG) [reply prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: Fixed, thanks

**Comment by chaodaiG** [comment]
testgrid config checker job failed due to a bug in transfigure image, reported in the other repo: https://github.com/kubernetes/test-infra/issues/17997

**Comment by chaodaiG** [comment]
Service account permission given

/unhold

**Pull request merged**